homeassistant/components/unifi/device_tracker.py

```diff
@@ -66,10 +67,65 @@
     EventKey.WIRELESS_GUEST_ROAM,
     EventKey.WIRELESS_GUEST_ROAMRADIO,
 )
+WIRELESS_DISCONNECTION = (
+    EventKey.WIRELESS_CLIENT_DISCONNECTED,
+    EventKey.WIRELESS_GUEST_DISCONNECTED,
+)
+
+
+@callback
+def async_client_allowed_fn(controller: UniFiController, obj_id: str) -> bool:
+    """Check if client is allowed."""
+    client = controller.api.clients[obj_id]
+    if not controller.option_track_clients:
+        return False
+
+    if client.mac not in controller.wireless_clients:
+        if not controller.option_track_wired_clients:
+            return False
 
+    elif (
+        client.essid
+        and controller.option_ssid_filter
+        and client.essid not in controller.option_ssid_filter
+    ):
+        return False
```

Comment:
Wonder if lines 87 - 94 could be replaced with:

```python
return client.essid
         and controller.option_ssid_filter
         and client.essid not in controller.option_ssid_filter
```

Reply:
Looking at them plainly you probably could, but there have been issues UniFi reporting values of things incorrectly, there is one old issue that wired devices when going offline gets marked as online wireless devices. This PR doesn't really change the logics of the client tracker so that would preferably be a future PR when this one has been vetted by users.

Reply:
I tried some changes in the allowed method but I think that made it less easy to read, and its complicated enough as it is so I will not do this suggestion